From 8e8c4e2d5665b3dfb1c447e69fb2066f9d7e004f Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 6 Sep 2022 07:43:16 +0800
Subject: [PATCH 15/60] Revert "drm/rockchip: Fix an error handling path
 rockchip_dp_probe()"

This reverts commit fc85cb33f6aae45872e4689d522089feae13d9ed.
---
 drivers/gpu/drm/rockchip/analogix_dp-rockchip.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/analogix_dp-rockchip.c b/drivers/gpu/drm/rockchip/analogix_dp-rockchip.c
index 512581698a1e0..ade2327a10e2c 100644
--- a/drivers/gpu/drm/rockchip/analogix_dp-rockchip.c
+++ b/drivers/gpu/drm/rockchip/analogix_dp-rockchip.c
@@ -398,15 +398,7 @@ static int rockchip_dp_probe(struct platform_device *pdev)
 	if (IS_ERR(dp->adp))
 		return PTR_ERR(dp->adp);
 
-	ret = component_add(dev, &rockchip_dp_component_ops);
-	if (ret)
-		goto err_dp_remove;
-
-	return 0;
-
-err_dp_remove:
-	analogix_dp_remove(dp->adp);
-	return ret;
+	return component_add(dev, &rockchip_dp_component_ops);
 }
 
 static int rockchip_dp_remove(struct platform_device *pdev)
-- 
2.25.1

